pytket/extensions/qiskit/backends/aer.py

```diff
@@ -256,6 +309,75 @@ def default_compilation_pass(
             optimisation_level, timeout
         )
 
+    def get_compiled_circuit(
+        self, circuit: Circuit, optimisation_level: int = 2, timeout: int = 300
+    ) -> Circuit:
+        """
+        Return a single circuit compiled with :py:meth:`default_compilation_pass`.
+
+        :param optimisation_level: Allows values of 0, 1, 2 or 3, with higher values
+            prompting more computationally heavy optimising compilation that
+            can lead to reduced gate count in circuits.
+        :type optimisation_level: int, optional
+        :param timeout: Only valid for optimisation level 3, gives a maximimum time
+            for running a single thread of the pass `GreedyPauliSimp`. Increase for
+            optimising larger circuits.
+        :type timeout: int, optional
+
+        :return: An optimised quantum circuit
+        :rtype: Circuit
+        """
+        return_circuit = circuit.copy()
+        if optimisation_level == 3 and circuit.n_gates_of_type(OpType.Barrier) > 0:
```

Comment:
Why is this check not in the other function `get_compiled_circuits` ?

Reply:
`get_compiled_circuits` calls this method so the check is done implicitly